server.go

```diff
@@ -520,24 +458,15 @@ func (srv *Server) serveUDP(l *net.UDPConn) error {
 	// deadline is not used here
 	for {
 		m, s, err := reader.ReadUDP(l, rtimeout)
-		srv.lock.RLock()
-		if !srv.started {
-			srv.lock.RUnlock()
-			return nil
-		}
-		srv.lock.RUnlock()
 		if err != nil {
-			continue
+			return err
```

Comment:
Is a closed listener the only way that this will err? cc @andrewtj @asergeyev

Reply:
Perhaps some resource running out might bubble up?

Reply:
I wondered about that too. The TCP listener has  this:
~~~
	for {
		rw, err := l.Accept()
		if err != nil {
			if neterr, ok := err.(net.Error); ok && neterr.Temporary() {
				continue
			}
			return err
		}
~~~